src/qonlinetts.cpp

```diff
@@ -161,6 +182,112 @@ QOnlineTts::Voice QOnlineTts::voice(const QString &voiceCode)
     return s_voiceCodes.key(voiceCode, NoVoice);
 }
 
+QPair<QOnlineTranslator::Language, QLocale::Country> QOnlineTts::region(const QString &regionCode)
+{
+    return s_regionCodes.key(regionCode, {QOnlineTranslator::NoLanguage, QLocale::AnyCountry});
+}
+
+QString QOnlineTts::regionName(QOnlineTranslator::Language language, QLocale::Country region)
+{
+    switch (language) {
+    case QOnlineTranslator::Bengali:
+        switch (region) {
+        case QLocale::Bangladesh:
+            return tr("Bangla (Bangladesh)");
+        case QLocale::India:
+            return tr("Bangla (India)");
+        default:
+            break;
+        }
+    case QOnlineTranslator::SimplifiedChinese:
+        switch (region) {
+        case QLocale::China:
+            return tr("Chinese, Mandarin (China)");
+        default:
+            break;
+        }
+    case QOnlineTranslator::English:
+        switch (region) {
+        case QLocale::Australia:
+            return tr("English (Australia)");
+        case QLocale::India:
+            return tr("English (India)");
+        case QLocale::UnitedKingdom:
+            return tr("English (United Kingdom)");
+        case QLocale::UnitedStates:
+            return tr("English (United States)");
+        default:
+            break;
+        }
+    case QOnlineTranslator::French:
+        switch (region) {
+        case QLocale::Canada:
+            return tr("French (Canada)");
+        case QLocale::France:
+            return tr("French (France)");
+        default:
+            break;
+        }
+    case QOnlineTranslator::German:
+        switch (region) {
+        case QLocale::Germany:
+            return tr("German (Germany)");
+        default:
+            break;
+        }
```

Comment:
I afraid you also need to add `break` here and in other simillar places :)
```suggestion
        }
        break;
```

Reply:
Got it done. :(

Reply:
No, no, you need to add it after each case with `QOnlineTranslator::*` too. It's because in C++ the default behavior is fallthough.

Reply:
Sorry, did you mean this?
```
    switch (language) {
    case QOnlineTranslator::Bengali:
        switch (region) {
        case QLocale::Bangladesh:
            return tr("Bangla (Bangladesh)");
        case QLocale::India:
            return tr("Bangla (India)");
        default:
            break;
        }
        break;
    case QOnlineTranslator::SimplifiedChinese:
        switch (region) {
        case QLocale::China:
            return tr("Chinese, Mandarin (China)");
        default:
            break;
        }
        break;
    case QOnlineTranslator::English:
        switch (region) {
        case QLocale::Australia:
            return tr("English (Australia)");
        case QLocale::India:
            return tr("English (India)");
        case QLocale::UnitedKingdom:
            return tr("English (United Kingdom)");
        case QLocale::UnitedStates:
            return tr("English (United States)");
        default:
            break;
        }
        break;
    case QOnlineTranslator::French:
        switch (region) {
        case QLocale::Canada:
            return tr("French (Canada)");
        case QLocale::France:
            return tr("French (France)");
        default:
            break;
        }
        break;
    case QOnlineTranslator::German:
        switch (region) {
        case QLocale::Germany:
            return tr("German (Germany)");
        default:
            break;
        }
        break;
    case QOnlineTranslator::Portuguese:
        switch (region) {
        case QLocale::Brazil:
            return tr("Portuguese (Brazil)");
        default:
            break;
        }
        break;
    case QOnlineTranslator::Spanish:
        switch (region) {
        case QLocale::Spain:
            return tr("Spanish (Spain)");
        case QLocale::UnitedStates:
            return tr("Spanish (United States)");
        default:
            break;
        }
        break;
    case QOnlineTranslator::Tamil:
        switch (region) {
        case QLocale::India:
            return tr("Tamil (India)");
        default:
            break;
        }
        break;
```
I posted the comment above and pushed at the same time, but strangely `git` did not immediately push.

Reply:
Yep!
A little noisy, but it's better to repeat keyword then a translation string. Will be merged automatically once CI pass.